python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -476,8 +477,39 @@ def runQuantum(
             task.runQuantum(butlerQC, inputRefs, outputRefs)
         except NoWorkFound as err:
             # Not an error, just an early exit.
-            _LOG.info("Task '%s' on quantum %s exited early: %s", task_node.label, quantum.dataId, str(err))
-            pass
+            _LOG.info(
+                "Task '%s' on quantum %s exited early with no work found: %s.",
+                task_node.label,
+                quantum.dataId,
+                str(err),
+            )
+        except AnnotatedPartialOutputsError as caught:
+            error: BaseException
+            if caught.__cause__ is None:
+                _LOG.warning(
+                    "Incorrect use of AnnotatedPartialOutputsError: no chained exception found.",
+                    task_node.label,
+                    quantum.dataId,
+                )
+                error = caught
+            else:
+                error = caught.__cause__
+            if self.raise_on_partial_outputs:
+                # This is a real edge case that required some experimentation:
+                # without 'from None' below, this raise would produce a
+                # "while one exception was being handled, another was raised"
+                # traceback involving AnnotatedPartialOutputsError.  With
+                # the 'from None', we get just the error chained to it, as
+                # desired.
+                raise error from None
+            else:
+                _LOG.warning(
+                    "Task '%s' on quantum %s exited with partial outputs; "
+                    "considering this a qualified success and proceeding.",
```

Comment:
That's a good way to phrase it. I do wonder if it shouldn't be `error` here, or `exception` maybe (we're allowed to log an `error` even if the task doesn't halt execution, right?), so it doesn't get lost in our massive field of warnings, but I guess that's up to DRP to decide.

Reply:
I'm not sure if we're _allowed_ to log `error` if we don't halt, but I don't like doing it because I think there's no common mental model of what it means.

Reply:
In fact, we explicitly can! Per the only place I know of that we've written any such recommendation:
https://developer.lsst.io/stack/logging.html#log-levels

`ERROR: for errors that may still allow the execution to continue.`